uaclient/entitlements/repo.py

```diff
@@ -95,11 +95,90 @@ def _perform_enable(self, silent: bool = False) -> bool:
         return True
 
     def _perform_disable(self, silent=False):
+        if self.purge:
```

Comment:
maybe we can add `self.purge and self.origin` instead ?

Reply:
We do the can_disable check, but yes - it doesn't hurt - can do.